tt_metal/hw/inc/ethernet/dataflow_api.h

```diff
@@ -110,13 +110,17 @@ void send_fd_packets() {
 }
 
 FORCE_INLINE
-void receive_fd_packets() {
+void wait_for_fd_packet() {
     // There may not be a valid cmd here, since DST router is always polling
     // This should only happen on cluster close
-    while (routing_info->fd_buffer_msgs_sent != 1 && routing_info->routing_enabled) {
+    while (routing_info->fd_buffer_msgs_sent != 1) {
```

Comment:
I don't think you can remove this `&& routing_info->routing_enabled`. If my thinking is correct, this wouldn't be an issue for the remote DST router that is receiving commands from local SRC. But for the local DST router getting sent stuff from remote SRC router for the return path, this will cause a hang.